platform/saivpp/vpplib/SwitchStateBaseRif.cpp

```diff
@@ -117,7 +123,140 @@ bool vpp_get_intf_name_for_prefix (
     }
 }
 
-bool SwitchStateBase::vpp_intf_get_prefix_entry (const std::string& intf_name, std::string& ip_prefix)
+void create_route_prefix (
+    sai_route_entry_t *route_entry,
+    vpp_ip_route_t *ip_route)
+{
+    const sai_ip_prefix_t *ip_address = &route_entry->destination;
+
+    switch (ip_address->addr_family) {
+    case SAI_IP_ADDR_FAMILY_IPV4:
+    {
+	struct sockaddr_in *sin =  &ip_route->prefix_addr.addr.ip4;
+
+	ip_route->prefix_addr.sa_family = AF_INET;
+	sin->sin_addr.s_addr = ip_address->addr.ip4;
+	ip_route->prefix_len = getPrefixLenFromAddrMask(reinterpret_cast<const uint8_t*>(&ip_address->mask.ip4), 4);
+
+	break;
+    }
+    case SAI_IP_ADDR_FAMILY_IPV6:
+    {
+	struct sockaddr_in6 *sin6 =  &ip_route->prefix_addr.addr.ip6;
+
+	ip_route->prefix_addr.sa_family = AF_INET6;
+	memcpy(sin6->sin6_addr.s6_addr, ip_address->addr.ip6, sizeof(sin6->sin6_addr.s6_addr));
+	ip_route->prefix_len = getPrefixLenFromAddrMask(ip_address->mask.ip6, 16);
+
+	break;
+    }
+    }
+}
+
+int configureLoopbackInterface (
+    bool isAdd,
+    const std::string &hostIfname,
+    const std::string &destinationIp,
+    int prefixLen) 
+{
+    SWSS_LOG_ENTER();
+
+    // Prepare the command
+    std::string command = "config interface ip ";
+    command += isAdd ? "add " : "remove ";
+    command += hostIfname + " " + destinationIp + "/" + std::to_string(prefixLen);
+ 
+    // Execute the command
+    FILE* pipe = popen(command.c_str(), "r");
```

Comment:
Please swss::exec() instead of creating your own. Please check SwitchStateBaseRif.cpp for an example.

Reply:
Fixed

Reply:
Please use swss::exec which does take care of the whole execution of commands.